src/LimitedAreaSimulations/neverworld_simulation.jl

```diff
@@ -29,17 +29,73 @@ using Oceananigans.Operators: Δzᶜᶜᶜ
 using Oceananigans.Coriolis: ActiveCellEnstrophyConservingScheme
 using Oceananigans.TurbulenceClosures: CATKEVerticalDiffusivity
 
-using ClimaOcean: CubicSplineFunction, u_bottom_drag, v_bottom_drag, u_immersed_bottom_drag, v_immersed_bottom_drag
+using ClimaOcean: CubicSplineFunction,
+                  u_bottom_drag, u_immersed_bottom_drag,
+                  v_bottom_drag, v_immersed_bottom_drag
+                  
 using ..VerticalGrids: stretched_vertical_cell_interfaces
 
+const c = Center()
+const f = Face()
+
+#####
+##### Geometry
+#####
+
+struct Point{T}
+    x :: T
+    y :: T
+end
+
+struct LineSegment{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+struct Line{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+distance(p₁::Point, p₂::Point) = sqrt((p₁.x - p₂.x)^2 + (p₁.y - p₂.y)^2)
+
+# https://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment
+function distance(point::Point, line::LineSegment)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+
+    # Line segment lengths
+    Δx = x₂ - x₁
+    Δy = y₂ - y₁
+
+    # Fractional increment to closest segment point
+    ϵ = ((x - x₁) * Δx + (y - y₁) * Δy) / (Δx^2 + Δy^2)
+    ϵ = clamp(ϵ, 0, 1)
+
+    # Closest segment point
+    x′ = x₁ + ϵ * Δx
+    y′ = y₁ + ϵ * Δy
+    
+    return sqrt((x′ - x)^2 + (y′ - y)^2)
+end
+
+function distance(point::Point, line::Line)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+    num = abs((x₂ - x₁) * (y₁ - y₀) - (y₂ - y₁) * (x₁ - x₀))
```

Comment:
what's `x₀` and `y₀`?

Reply:
typo that should be `x, y`

Reply:
but maybe should delete this code cause we dont need it

Reply:
doesn't the bathymetry constructor use it?

Reply:
no

Reply:
this is for `Line`, the constructor uses `LineSegment`